Patch-Source: https://github.com/nginx-shib/nginx-http-shibboleth/commit/99d72dc82c56e8fc22e97dfaa9b61882361565f4
--
From 99d72dc82c56e8fc22e97dfaa9b61882361565f4 Mon Sep 17 00:00:00 2001
From: Heiko Jansen <jansen@hbz-nrw.de>
Date: Thu, 18 Aug 2022 17:30:14 +0200
Subject: [PATCH] bugfix: nginx crash when accessing uninitialized pointer

This copies code contributed to ngx_headers_more by @zhuizhuhaomeng
in openresty/headers-more-nginx-module@d502e41

I omitted the PCRE2-related build flag added in that same commit because
I do not understand the reason for that.
---
 ngx_http_shibboleth_module.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ngx_http_shibboleth_module.c b/ngx_http_shibboleth_module.c
index cbfb37d..d211a11 100644
--- a/ngx_http_shibboleth_module.c
+++ b/ngx_http_shibboleth_module.c
@@ -763,6 +763,9 @@ ngx_http_set_header_helper(ngx_http_request_t *r, ngx_http_shib_request_header_v
 
     h->key = hv->key;
     h->value = *value;
+#if defined(nginx_version) && nginx_version >= 1023000
+    h->next = NULL;
+#endif
 
     h->lowcase_key = ngx_pnalloc(r->pool, h->key.len);
     if (h->lowcase_key == NULL) {
